iceoryx_utils/testutils/timing_test.hpp

```diff
@@ -0,0 +1,142 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include <atomic>
+#include <cstdint>
+#include <functional>
+#include <string>
+
+/// @brief This headers provides TIMING_TEST unit test infrastructure. The idea
+///         is that a timing test is running multiple times and if in one of the
+///         repetitions all results of the test are successful then the timing
+///         test itself is successful.
+///
+///        You can get to know them in praxis in the test_posix_timer.cpp unit test
+///        for instance.
+///
+/// @code
+/// class MyClass_test : public Test {};
+///
+/// TIMING_TEST_F(MyClass_test, WaitForSleep, Repeat(3), [&]{
+///   std::atomic_bool threadFinished{false};
+///   std::thread t([&]{ sleep(2); threadFinished = true; });
+///
+///   TIMING_TEST_EXPECT_FALSE(threadFinished.load());
+///   sleep(1);
+///   TIMING_TEST_EXPECT_FALSE(threadFinished.load());
+///   sleep(2);
+///   TIMING_TEST_EXPECT_TRUE(threadFinished.load());
+///   t.join();
+/// });
+/// @endcode
+///
+/// @details
+///     Available testing verificators
+///        TIMING_TEST_EXPECT_TRUE(value);
+///        TIMING_TEST_EXPECT_FALSE(value);
+///        TIMING_TEST_ASSERT_TRUE(value);
+///        TIMING_TEST_ASSERT_FALSE(value);
+///
+///     Available test types
+///         TIMING_TEST_F -> maps to TEST_F
+///         TIMING_TEST_P -> maps to TEST_P
+///
+///     If you would like to disable timing tests you can start your unit
+///     test like:
+///      # ./myUnitTest --gtest_filter="-*TimingTest*"
+///
+///     Or if you would like only timing test you can use the following
+///     approach:
+///      # ./myUnitTest --gtest_filter="*TimingTest*"
+
+#define TIMING_TEST_CONSTRUCT(Name, Case, Repetitions, Test, GTestType)                                                \
+    GTestType(Name, TimingTest_##Case)                                                                                 \
+    {                                                                                                                  \
+        std::atomic_bool timingTestResult{true};                                                                       \
+        std::string errorMessages;                                                                                     \
+        bool testResult = iox::testutils::performingTimingTest(Test, Repetitions, timingTestResult);                   \
+        EXPECT_TRUE(testResult);                                                                                       \
+        if (!testResult)                                                                                               \
+        {                                                                                                              \
+            std::cout << "\n" << errorMessages << std::endl;                                                           \
```

Comment:
I'm curious why here two times newline is used in different shapes. As far as i understood std::endl is flushing the buffer and \n not. Is this here the reason for it?

Reply:
The \n in the beginning adds a new line which improves the readability quiet a bit. Otherwise you barely see where the normal gtest message ends and the timing error trace begins.